akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2StreamHandling.scala

```diff
@@ -96,6 +97,15 @@ private[http2] trait Http2StreamHandling { self: GraphStageLogic with LogHelper
 
   def activeStreamCount(): Int = streamStates.size
 
+  /**
+   * This method signals the Http2StreamHandling that the enclosing stage (Http2Demux) is
+   * completing because the userland inlet was finished. Do not invoke this when the cause
+   * is the connection is closed or in case of failure in any of the inlets.
+   */
+  def onUserUpstreamFinish(): Unit = {
+    userUpstreamFinishing = true
+  }
```

Comment:
Does it need to be a method when we only call it from one place and all it does is toggling a boolean named the same? (explanation could go on field instead if important)